src/openforms/contrib/objects_api/validators.py

```diff
@@ -78,3 +91,43 @@ def validate_document_type_references(
 
     if errors:
         raise ValidationError(errors)
+
+
+def validate_object_ownership(form: Form, session, initial_data_reference: str) -> None:
+    auth_info = session.get(FORM_AUTH_SESSION_KEY)
+    if not auth_info:
+        raise PermissionDenied("Cannot pass data reference as anonymous user")
+
+    object = None
+    for backend in form.registration_backends.filter(backend="objects_api"):
+        if not backend.options:
+            continue
+
+        api_group = ObjectsAPIGroupConfig.objects.filter(
+            pk=backend.options.get("objects_api_group")
+        ).first()
+        if not api_group:
+            continue
+
+        try:
+            with get_objects_client(api_group) as client:
+                try:
+                    object = client.get_object(initial_data_reference)
+                    break
+                except RequestException:
+                    logger.exception(
+                        "Something went wrong while trying to retrieve "
+                        "object for product prefill permission check"
```

Comment:
Will this be specific to the product prefill exception?I had in mind that checking the owner of the object could be useful to other things in the future as well.

Reply:
Hmm yeah you're right, I will change it to not mention prefill